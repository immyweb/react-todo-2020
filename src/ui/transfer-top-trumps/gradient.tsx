import * as React from 'react';

import { GradientContainer, GradientGroup, GradientLine } from './styles';

export const Gradient: React.SFC<{
  colour: string;
  size?: number;
  top?: number;
  height?: number;
  opacity?: number;
}> = ({ colour, size, top, height, opacity }) => {
  return (
    <GradientContainer size={size} top={top} opacity={opacity}>
      <GradientGroup width={'60%'}>
        <GradientLine direction={'right'} teamColour={colour} height={height} />
        <GradientLine direction={'left'} teamColour={colour} height={height} />
      </GradientGroup>
      <GradientGroup width={'100%'}>
        <GradientLine direction={'right'} teamColour={colour} height={height} />
        <GradientLine direction={'left'} teamColour={colour} height={height} />
      </GradientGroup>
      <GradientGroup width={'60%'}>
        <GradientLine direction={'right'} teamColour={colour} height={height} />
        <GradientLine direction={'left'} teamColour={colour} height={height} />
      </GradientGroup>
    </GradientContainer>
  );
};
