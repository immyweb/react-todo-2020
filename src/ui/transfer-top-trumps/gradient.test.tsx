import * as React from 'react';
import { shallow } from 'enzyme';

import { Gradient } from './gradient';

describe('<Gradient />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Gradient colour={'#ffffff'} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('can customise colour, size, height and opacity', () => {
    const wrapper = shallow(<Gradient colour={'#c40000'} size={5} height={100} opacity={0.2} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('can offset top position', () => {
    const wrapper = shallow(<Gradient colour={'#c40000'} top={50} />);
    expect(wrapper).toMatchSnapshot();
  });
});
