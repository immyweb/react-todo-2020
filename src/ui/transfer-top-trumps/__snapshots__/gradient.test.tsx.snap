// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Gradient /> can customise colour, size, height and opacity 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    colour="#c40000"
    height={100}
    opacity={0.2}
    size={5}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <glamorous(div)
          width="60%"
        >
          <glamorous(div)
            direction="right"
            height={100}
            teamColour="#c40000"
          />
          <glamorous(div)
            direction="left"
            height={100}
            teamColour="#c40000"
          />
        </glamorous(div)>,
        <glamorous(div)
          width="100%"
        >
          <glamorous(div)
            direction="right"
            height={100}
            teamColour="#c40000"
          />
          <glamorous(div)
            direction="left"
            height={100}
            teamColour="#c40000"
          />
        </glamorous(div)>,
        <glamorous(div)
          width="60%"
        >
          <glamorous(div)
            direction="right"
            height={100}
            teamColour="#c40000"
          />
          <glamorous(div)
            direction="left"
            height={100}
            teamColour="#c40000"
          />
        </glamorous(div)>,
      ],
      "opacity": 0.2,
      "size": 5,
      "top": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <glamorous(div)
              direction="right"
              height={100}
              teamColour="#c40000"
            />,
            <glamorous(div)
              direction="left"
              height={100}
              teamColour="#c40000"
            />,
          ],
          "width": "60%",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "direction": "right",
              "height": 100,
              "teamColour": "#c40000",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "direction": "left",
              "height": 100,
              "teamColour": "#c40000",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <glamorous(div)
              direction="right"
              height={100}
              teamColour="#c40000"
            />,
            <glamorous(div)
              direction="left"
              height={100}
              teamColour="#c40000"
            />,
          ],
          "width": "100%",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "direction": "right",
              "height": 100,
              "teamColour": "#c40000",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "direction": "left",
              "height": 100,
              "teamColour": "#c40000",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <glamorous(div)
              direction="right"
              height={100}
              teamColour="#c40000"
            />,
            <glamorous(div)
              direction="left"
              height={100}
              teamColour="#c40000"
            />,
          ],
          "width": "60%",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "direction": "right",
              "height": 100,
              "teamColour": "#c40000",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "direction": "left",
              "height": 100,
              "teamColour": "#c40000",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <glamorous(div)
            width="60%"
          >
            <glamorous(div)
              direction="right"
              height={100}
              teamColour="#c40000"
            />
            <glamorous(div)
              direction="left"
              height={100}
              teamColour="#c40000"
            />
          </glamorous(div)>,
          <glamorous(div)
            width="100%"
          >
            <glamorous(div)
              direction="right"
              height={100}
              teamColour="#c40000"
            />
            <glamorous(div)
              direction="left"
              height={100}
              teamColour="#c40000"
            />
          </glamorous(div)>,
          <glamorous(div)
            width="60%"
          >
            <glamorous(div)
              direction="right"
              height={100}
              teamColour="#c40000"
            />
            <glamorous(div)
              direction="left"
              height={100}
              teamColour="#c40000"
            />
          </glamorous(div)>,
        ],
        "opacity": 0.2,
        "size": 5,
        "top": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <glamorous(div)
                direction="right"
                height={100}
                teamColour="#c40000"
              />,
              <glamorous(div)
                direction="left"
                height={100}
                teamColour="#c40000"
              />,
            ],
            "width": "60%",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "direction": "right",
                "height": 100,
                "teamColour": "#c40000",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "direction": "left",
                "height": 100,
                "teamColour": "#c40000",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <glamorous(div)
                direction="right"
                height={100}
                teamColour="#c40000"
              />,
              <glamorous(div)
                direction="left"
                height={100}
                teamColour="#c40000"
              />,
            ],
            "width": "100%",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "direction": "right",
                "height": 100,
                "teamColour": "#c40000",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "direction": "left",
                "height": 100,
                "teamColour": "#c40000",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <glamorous(div)
                direction="right"
                height={100}
                teamColour="#c40000"
              />,
              <glamorous(div)
                direction="left"
                height={100}
                teamColour="#c40000"
              />,
            ],
            "width": "60%",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "direction": "right",
                "height": 100,
                "teamColour": "#c40000",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "direction": "left",
                "height": 100,
                "teamColour": "#c40000",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Gradient /> can offset top position 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    colour="#c40000"
    top={50}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <glamorous(div)
          width="60%"
        >
          <glamorous(div)
            direction="right"
            teamColour="#c40000"
          />
          <glamorous(div)
            direction="left"
            teamColour="#c40000"
          />
        </glamorous(div)>,
        <glamorous(div)
          width="100%"
        >
          <glamorous(div)
            direction="right"
            teamColour="#c40000"
          />
          <glamorous(div)
            direction="left"
            teamColour="#c40000"
          />
        </glamorous(div)>,
        <glamorous(div)
          width="60%"
        >
          <glamorous(div)
            direction="right"
            teamColour="#c40000"
          />
          <glamorous(div)
            direction="left"
            teamColour="#c40000"
          />
        </glamorous(div)>,
      ],
      "opacity": undefined,
      "size": undefined,
      "top": 50,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <glamorous(div)
              direction="right"
              teamColour="#c40000"
            />,
            <glamorous(div)
              direction="left"
              teamColour="#c40000"
            />,
          ],
          "width": "60%",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "direction": "right",
              "height": undefined,
              "teamColour": "#c40000",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "direction": "left",
              "height": undefined,
              "teamColour": "#c40000",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <glamorous(div)
              direction="right"
              teamColour="#c40000"
            />,
            <glamorous(div)
              direction="left"
              teamColour="#c40000"
            />,
          ],
          "width": "100%",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "direction": "right",
              "height": undefined,
              "teamColour": "#c40000",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "direction": "left",
              "height": undefined,
              "teamColour": "#c40000",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <glamorous(div)
              direction="right"
              teamColour="#c40000"
            />,
            <glamorous(div)
              direction="left"
              teamColour="#c40000"
            />,
          ],
          "width": "60%",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "direction": "right",
              "height": undefined,
              "teamColour": "#c40000",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "direction": "left",
              "height": undefined,
              "teamColour": "#c40000",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <glamorous(div)
            width="60%"
          >
            <glamorous(div)
              direction="right"
              teamColour="#c40000"
            />
            <glamorous(div)
              direction="left"
              teamColour="#c40000"
            />
          </glamorous(div)>,
          <glamorous(div)
            width="100%"
          >
            <glamorous(div)
              direction="right"
              teamColour="#c40000"
            />
            <glamorous(div)
              direction="left"
              teamColour="#c40000"
            />
          </glamorous(div)>,
          <glamorous(div)
            width="60%"
          >
            <glamorous(div)
              direction="right"
              teamColour="#c40000"
            />
            <glamorous(div)
              direction="left"
              teamColour="#c40000"
            />
          </glamorous(div)>,
        ],
        "opacity": undefined,
        "size": undefined,
        "top": 50,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <glamorous(div)
                direction="right"
                teamColour="#c40000"
              />,
              <glamorous(div)
                direction="left"
                teamColour="#c40000"
              />,
            ],
            "width": "60%",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "direction": "right",
                "height": undefined,
                "teamColour": "#c40000",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "direction": "left",
                "height": undefined,
                "teamColour": "#c40000",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <glamorous(div)
                direction="right"
                teamColour="#c40000"
              />,
              <glamorous(div)
                direction="left"
                teamColour="#c40000"
              />,
            ],
            "width": "100%",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "direction": "right",
                "height": undefined,
                "teamColour": "#c40000",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "direction": "left",
                "height": undefined,
                "teamColour": "#c40000",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <glamorous(div)
                direction="right"
                teamColour="#c40000"
              />,
              <glamorous(div)
                direction="left"
                teamColour="#c40000"
              />,
            ],
            "width": "60%",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "direction": "right",
                "height": undefined,
                "teamColour": "#c40000",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "direction": "left",
                "height": undefined,
                "teamColour": "#c40000",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Gradient /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    colour="#ffffff"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <glamorous(div)
          width="60%"
        >
          <glamorous(div)
            direction="right"
            teamColour="#ffffff"
          />
          <glamorous(div)
            direction="left"
            teamColour="#ffffff"
          />
        </glamorous(div)>,
        <glamorous(div)
          width="100%"
        >
          <glamorous(div)
            direction="right"
            teamColour="#ffffff"
          />
          <glamorous(div)
            direction="left"
            teamColour="#ffffff"
          />
        </glamorous(div)>,
        <glamorous(div)
          width="60%"
        >
          <glamorous(div)
            direction="right"
            teamColour="#ffffff"
          />
          <glamorous(div)
            direction="left"
            teamColour="#ffffff"
          />
        </glamorous(div)>,
      ],
      "opacity": undefined,
      "size": undefined,
      "top": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <glamorous(div)
              direction="right"
              teamColour="#ffffff"
            />,
            <glamorous(div)
              direction="left"
              teamColour="#ffffff"
            />,
          ],
          "width": "60%",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "direction": "right",
              "height": undefined,
              "teamColour": "#ffffff",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "direction": "left",
              "height": undefined,
              "teamColour": "#ffffff",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <glamorous(div)
              direction="right"
              teamColour="#ffffff"
            />,
            <glamorous(div)
              direction="left"
              teamColour="#ffffff"
            />,
          ],
          "width": "100%",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "direction": "right",
              "height": undefined,
              "teamColour": "#ffffff",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "direction": "left",
              "height": undefined,
              "teamColour": "#ffffff",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <glamorous(div)
              direction="right"
              teamColour="#ffffff"
            />,
            <glamorous(div)
              direction="left"
              teamColour="#ffffff"
            />,
          ],
          "width": "60%",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "direction": "right",
              "height": undefined,
              "teamColour": "#ffffff",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "direction": "left",
              "height": undefined,
              "teamColour": "#ffffff",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <glamorous(div)
            width="60%"
          >
            <glamorous(div)
              direction="right"
              teamColour="#ffffff"
            />
            <glamorous(div)
              direction="left"
              teamColour="#ffffff"
            />
          </glamorous(div)>,
          <glamorous(div)
            width="100%"
          >
            <glamorous(div)
              direction="right"
              teamColour="#ffffff"
            />
            <glamorous(div)
              direction="left"
              teamColour="#ffffff"
            />
          </glamorous(div)>,
          <glamorous(div)
            width="60%"
          >
            <glamorous(div)
              direction="right"
              teamColour="#ffffff"
            />
            <glamorous(div)
              direction="left"
              teamColour="#ffffff"
            />
          </glamorous(div)>,
        ],
        "opacity": undefined,
        "size": undefined,
        "top": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <glamorous(div)
                direction="right"
                teamColour="#ffffff"
              />,
              <glamorous(div)
                direction="left"
                teamColour="#ffffff"
              />,
            ],
            "width": "60%",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "direction": "right",
                "height": undefined,
                "teamColour": "#ffffff",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "direction": "left",
                "height": undefined,
                "teamColour": "#ffffff",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <glamorous(div)
                direction="right"
                teamColour="#ffffff"
              />,
              <glamorous(div)
                direction="left"
                teamColour="#ffffff"
              />,
            ],
            "width": "100%",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "direction": "right",
                "height": undefined,
                "teamColour": "#ffffff",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "direction": "left",
                "height": undefined,
                "teamColour": "#ffffff",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <glamorous(div)
                direction="right"
                teamColour="#ffffff"
              />,
              <glamorous(div)
                direction="left"
                teamColour="#ffffff"
              />,
            ],
            "width": "60%",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "direction": "right",
                "height": undefined,
                "teamColour": "#ffffff",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "direction": "left",
                "height": undefined,
                "teamColour": "#ffffff",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
